lib/merkle_tree.ex

```diff
@@ -84,14 +107,25 @@ defmodule MerkleTree do
     _build(parents, hash_function, height)
   end
 
-  @spec is_power_of_n(pos_integer, pos_integer) :: boolean
-  defp is_power_of_n(n, x) do
-    (:math.log2(x)/:math.log2(n)) |> is_integer_float
+  defp fill_blocks(blocks, default, nil) when default != nil do
+    amout_elements = Enum.count(blocks)
+    expected_amout = :math.pow(2, :math.ceil(:math.log2(amout_elements)))
+    blocks ++ List.duplicate(default, trunc(expected_amout - amout_elements))
   end
 
-  @spec is_integer_float(float) :: boolean
-  defp is_integer_float(x) do
-    (Float.ceil x) == (Float.floor x)
+  defp fill_blocks(blocks, default, height) when default != nil do
+    amout_elements = Enum.count(blocks)
+    expected_amout = :math.pow(2, height)
+    fill_elements = expected_amout - amout_elements
+    if fill_elements < 0, do: raise(MerkleTree.ArgumentError)
+    blocks ++ List.duplicate(default, trunc(expected_amout - amout_elements))
   end
 
+  defp fill_blocks(blocks, _, _) when blocks != [] do
+    amout_elements = Enum.count(blocks)
+
+    if :math.pow(2, :math.ceil(:math.log2(amout_elements))) != amout_elements,
```

Comment:
`:math.pow(2, :math.ceil(:math.log2(amout_elements)))` - extract to variable `required_leaves_count` ?